tests/integration/test_nan_reduction.py

```diff
@@ -0,0 +1,259 @@
+
+        # relax criteria when doing floating ops with nans excluded
+        if func_np.__name__ == "nanprod" or func_np.__name__ == "nansum":
+            assert np.allclose(out_num, out_np, rtol=1e-4)
+        else:
+            assert np.array_equal(out_num, out_np)
+
+    @pytest.mark.parametrize("func_name", NAN_FUNCS)
+    @pytest.mark.parametrize("ndim", range(1, LEGATE_MAX_DIM + 1))
+    def test_out(self, func_name, ndim):
+        """This test checks that the out argument is updated with the
+        output"""
+
+        shape = (3,) * ndim
+        in_np = np.random.random(shape)
+        in_num = num.array(in_np)
+
+        func_num = getattr(num, func_name)
+        func_np = getattr(np, func_name)
+
+        axes = list(range(0, ndim - 1))
+        for axis in axes:
+            _shape = list(shape)
+            _shape[axis] = 1
+
+            out_num = num.empty(_shape)
+            func_num(in_num, out=out_num, axis=axis, keepdims=True)
+
+            out_np = np.empty(_shape)
+            func_np(in_np, out=out_np, axis=axis, keepdims=True)
+
+            assert np.allclose(out_num, out_np, rtol=1e-4)
+
+    @pytest.mark.parametrize("func_name", ("nanmin", "nanmax"))
+    @pytest.mark.parametrize("ndim", range(1, LEGATE_MAX_DIM + 1))
+    def test_all_nan(self, func_name, ndim):
+        """This test checks if we comply with the expected behavior when
+        the array contains only NaNs. The expected behavior is to
+        raise a ValueError.
+        """
+
+        shape = (3,) * ndim
+        in_num = num.zeros(shape)
+        in_num.fill(num.nan)
+
+        func_num = getattr(num, func_name)
+        out = 0.0
+        with pytest.warns(RuntimeWarning):
+            out = func_num(in_num)
+            assert np.isnan(out).any()
+
+    @pytest.mark.parametrize("func_name", ("nanmin", "nanmax"))
+    @pytest.mark.parametrize("ndim", range(2, LEGATE_MAX_DIM + 1))
+    @pytest.mark.parametrize("dtype", (np.float16, np.float32, np.float64))
+    @pytest.mark.parametrize("keepdims", [True, False])
+    def test_slice_nan(self, func_name, ndim, dtype, keepdims):
+        """This test checks if we comply with the expected behavior when
+        a slice contains only NaNs. The expected behavior is to issue a
+        RuntimeWarning and return NaN for that slice. Both are tested here.
+        """
+        shape = (3,) * ndim
+
+        in_np = np.random.random(shape)
+        in_num = num.array(in_np).astype(dtype)
+
+        in_num[0, ...] = num.nan
+        in_np[0, ...] = np.nan
+
+        func_num = getattr(num, func_name)
+        func_np = getattr(np, func_name)
+        func_num(in_num)
+
+        with pytest.warns(RuntimeWarning):
+            out_np = func_np(in_np, axis=1, keepdims=keepdims)
+            assert np.isnan(out_np).any()
+
+        with pytest.warns(RuntimeWarning):
+            out_num = func_num(in_num, axis=1, keepdims=keepdims)
+            assert np.isnan(out_num).any()
```

Comment:
This is the pattern we have adopted for these kinds of tests that compare exceptions (here and anywhere else applicable)
```suggestion
        expected_exc = RuntimeWarning
        
        with pytest.warns(expected_exc):
            out_np = func_np(in_np, axis=1, keepdims=keepdims)
            assert np.isnan(out_np).any()

        with pytest.warns(expected_exc):
            out_num = func_num(in_num, axis=1, keepdims=keepdims)
            assert np.isnan(out_num).any()
```